torch/_functorch/aot_autograd.py

```diff
@@ -737,6 +738,13 @@ def _dup_fake_script_obj(fake_flat_args):
                             static_input_indices=fw_metadata.static_input_indices,
                         )
 
+        if aot_config.sticky_cache is not None:
+            if any(
+                info.mutation_type != MutationType.NOT_MUTATED
```

Comment:
let me know if you want more review on the AOTAutograd bits (we might want to think more about how this interacts with the existing AOT warm cache today).

One comment is that input mutations are probably fine for sticky cache, as long as the input mutation is captured inside of the graph (the "bad" case is if the input mutation is forced to run outside of the graph, in an opaque AOTAutograd epilogue). Almost all input mutations at this point are captured in-graph today